### PR TITLE
[BotBuilder-AI] Make BotBuilder Ai web compatible

### PR DESCRIPTION
**Pull request by VictorGrycuk**
Fixes issue number 620

## Description
Remove NodeJS dependent libraries.

## Specific Changes
- Replace `request-promise-native` with `node-fetch`
- `@azure/cognitiveservices-luis-runtime` was fix in pr # 844

### CODE REVIEW

File: libraries/botbuilder-ai/src/qnaMaker.ts [hunk 1]
```diff
@@ -464,18 +464,19 @@ export class QnAMaker implements QnAMakerTelemetryClient {
...
                 ...queryOptions
-            }
+            })
         });
-
-        return this.formatQnaResult(qnaResult);
+        
+        const qnaResultJson: any = await qnaResult.json();
+        
+        return this.formatQnaResult(qnaResultJson);
```
Comment (GasparAcevedoZainSouthworks) [inline libraries/botbuilder-ai/src/qnaMaker.ts]: Can we merged both lines (477-479) into a single one,  with something like this:

```
return this.formatQnaResult(await qnaResult.json());
```

File: libraries/botbuilder-ai/src/qnaMaker.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import { Activity, TurnContext, BotTelemetryClient, NullTelemetryClient } from '
 import * as entities from 'html-entities';
 import * as os from 'os';
 const pjson: any = require('../package.json');
-import * as request from 'request-promise-native';
+const request = (new Function('require', 'if (!this.hasOwnProperty("fetch")) { return require("node-fetch"); } else { return this.fetch; }'))(require);
```
Comment (GasparAcevedoZainSouthworks) [inline libraries/botbuilder-ai/src/qnaMaker.ts]: Shouldn't we rename the `request` variable into something like `fetch`?

File: libraries/botbuilder-ai/src/qnaMaker.ts (hunk 1 above)
Comment (VictorGrycuk) [reply libraries/botbuilder-ai/src/qnaMaker.ts]: I have tried that approach, but it throws an exception.

File: libraries/botbuilder-ai/src/qnaMaker.ts (hunk 2 above)
Comment (VictorGrycuk) [reply libraries/botbuilder-ai/src/qnaMaker.ts]: I tried to leave it as close as possible to the original code to avoid confusion.

**Comment by VictorGrycuk** [comment]
This PR will be closed to open one with a more general branch name.

**Pull request closed without merging**